qttypes/src/lib.rs

```diff
@@ -640,23 +641,105 @@ impl std::fmt::Debug for QString {
+        cpp!(unsafe [self as "const QVariant*", target_type_id as "int"] -> bool as "bool" {
+            return self->canConvert(target_type_id);
+        })
+    }
+
+    // TODO: for this to work, we need QMetaType be available in qttypes package.
+    // TODO: It would create a dependency cycle at this point.
+    // /// Wrapper around [`canConvert()`][method] method.
+    // ///
+    // /// [method]: https://doc.qt.io/qt-5/qvariant.html#canConvert-1
+    // pub fn can_convert<T: QMetaType>(&self) -> bool {
+    //     self.can_convert(T::id())
+    // }
+
+    /// Wrapper around [`clear()`][method] method.
+    ///
+    /// [method]: https://doc.qt.io/qt-5/qvariant.html#clear
+    pub fn clear(&mut self) {
+        cpp!(unsafe [self as "QVariant*"] {
+            self->clear();
+        });
+    }
+
+    /// Wrapper around [`convert(int targetTypeId)`][method] method.
+    ///
+    /// [method]: https://doc.qt.io/qt-5/qvariant.html#convert
+    pub fn convert(&mut self, target_type_id: i32) -> bool {
+        cpp!(unsafe [self as "QVariant*", target_type_id as "int"] -> bool as "bool" {
+            return self->convert(target_type_id);
+        })
+    }
+
+    // TODO?: [static] template <typename Types> QVariant QVariant::fromStdVariant(const std::variant<Types...> &value)
+    // TODO: [static] template <typename T> QVariant QVariant::fromValue(const T &value)
+    // TODO: template <typename T> void QVariant::setValue(const T &value)
+    // either needs compile-time resolution of QMetaType for the T, which requires QMetaType to be
+    // available is this package.
+    // /// Wrapper around [`setValue(const T &value)`][method] method.
+    // ///
+    // /// [method]: https://doc.qt.io/qt-5/qvariant.html#setValue
+    // pub fn set_value<T: QMetaType>(&mut self, value: &T) {
+    //     // we have to convert value to some runtime-static type in Rust,
+    //     // because generic T won't fit through FFI boundaries.
+    //     let var = value.to_qvariant();
+    //     cpp!(unsafe [self as "QVariant*", value as "QVariant"] {
+    //         self->setValue(value);
+    //     });
+    // }
+
+    /// Wrapper around [`isNull()`][method] method.
+    ///
+    /// [method]: https://doc.qt.io/qt-5/qvariant.html#isNull
+    pub fn is_null(&self) -> bool {
+        cpp!(unsafe [self as "const QVariant*"] -> bool as "bool" {
+            return self->isNull();
+        })
+    }
+
+    /// Wrapper around [`isValid()`][method] method.
+    ///
+    /// [method]: https://doc.qt.io/qt-5/qvariant.html#isValid
+    pub fn is_valid(&self) -> bool {
+        cpp!(unsafe [self as "const QVariant*"] -> bool as "bool" {
+            return self->isValid();
         })
     }
 
+    // XXX: for setValue() see fromValue() comment above.
+
+    /// Wrapper around [`swap(QVariant &other)`][method] method.
+    ///
+    /// [method]: https://doc.qt.io/qt-5/qvariant.html#swap
+    pub fn swap(&mut self, other: &mut QVariant) {
```

Comment:
This function also doesn't make sense in rust. One would use  the std::mem::swap function.

Reply:
yeah, kinda thought about std::mem::swap as well.  It was so weird to work with reference to a pointer passed through FFI, that I even wrote a test to make sure everything matched up.